sale_order_line_initial_quantity/models/sale.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2021 Akretion France (http://www.akretion.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    initial_qty_changed = fields.Boolean(
+        string="Initial Quantity Changed",
+        compute="_compute_initial_qty_changed",
+        store=True,
+    )
+
+    @api.depends("order_line", "order_line.product_uom_qty")
+    def _compute_initial_qty_changed(self):
+        for order in self:
+            order.initial_qty_changed = False
+            if order.state == "sale":
+                lines = order.order_line.filtered(
+                    lambda l: l.product_uom_qty != l.product_uom_initial_qty
+                )
+                if lines:
+                    order.initial_qty_changed = True
+
+    def write(self, values):
+        for order in self:
+            if "state" in values and values["state"] == "sale":
+                for line in order.order_line:
+                    line.product_uom_initial_qty = line.product_uom_qty
```

Comment:
Wrong  behaviour, if you have 3 products when you validate the sale order and you validated it and then you change to 2 and after to 1, the initial product qty will be 2 instead of 3